controllers/replication/secret_controller.go

```diff
@@ -312,6 +324,9 @@ func requiresUpdate(source, dest client.Object) bool {
 }
 
 func syncSecrets(src, dest *corev1.Secret) {
+	// TODO: @Michael what does this do, and do I need to worry about differences in the names between the source and destination?
```

Comment:
It enforces that the destination ObjectMeta is the same as the source ObjectMeta, as DeepCopyInto would not remove additional stuff from the modified destination object.

Reply:
Can I rename this as "removeDCSpecificMetadata" or something similar? Just so future implementors have a better idea what's happening.

Reply:
This function isn't about removing anything Datacenter specific. Sure, we filter out cassandra.datastax.com annotations (also cluster, rack, etc) from the target since these are restricted annotations / labels, but that's not all it does.

It also ensures that the annotation and label values set in the original one are having higher priority than the ones in the target secret.

So lets say the original one has labels A=B, C=D, but someone has modified the target one to have A=E, C=D, F=J. We will overwrite the targets to A=B, C=D, F=J set. We're not removing some existing ones from the target (there are some important ones), just ensuring that the same label can't have different value.

Thus, it's really about syncing metadata and that's where the name of the function comes from. The testSyncSecrets explains the syncing from one target to another.